### PR TITLE
Add ST X-NUCLEO-EEPRMA2 EEPROM shield and a EEPROM driver sample

### PR DESCRIPTION
**Pull request by str4t0m**
This PR adds the ST X-NUCLEO-EEPRMA2 as a new Arduino shield and adds a minimal sample to demonstrate the EEPROM driver api.

### CODE REVIEW

**Comment by str4t0m** [comment]
I see from @henrikbrixandersen as eeprom maintainer no review was requested.
Could you have a look at this?

File: samples/drivers/eeprom/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    harness: console
+    harness_config:
+      type: one_line
+      regex:
+        - "Reset the MCU to see the inceasing boot counter."
+
+tests:
+  sample.drivers.eeprom:
+    tags: eeprom
+    platform_allow: native_posix
```
Comment (erwango) [inline samples/drivers/eeprom/sample.yaml:16]: Adding a variant with shield would be nice. Not blocking

**Review by erwango — APPROVED** [review]
Looks OK from the shield point of view.
Need EEPROM maintainer to confirm for sample

**Comment by str4t0m** [comment]
I have added a "#include <freq.h>" in the shield overlay as most platforms with the exception of ST haven't included this themself.
And added a test configuration for the shield in the eeprom sample.

File: samples/drivers/eeprom/sample.yaml (hunk 1 above)
Comment (str4t0m) [reply samples/drivers/eeprom/sample.yaml:18]: Added a configuration, but I am not sure about the integration platforms.
I think most boards are already filtered due to console harness?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
A couple of minor nits, otherwise looks great.

File: samples/drivers/eeprom/sample.yaml (hunk 1 above)
Comment (str4t0m) [reply samples/drivers/eeprom/sample.yaml:16]: Only native_posix is now listed as integration platform.

File: samples/drivers/eeprom/README.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+as follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/drivers/eeprom
+   :board: nrf52840dk_nrf52840
+   :goals: build
+   :shield: x_nucleo_eeprma2
+   :compact:
+
+
```
[the remark this answers is not in the document]
Comment (str4t0m) [reply samples/drivers/eeprom/README.rst]: Done

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@str4t0m If you could you help verify #35743 on this shield I would be grateful.

**Comment by str4t0m** [comment]
The eeprom tests still pass on the shield.

I wonder whether I should add an eeprom-1 alias for the eeprom4 node to the shield, such that one I2C and one SPI device are active?

**Comment by henrikbrixandersen** [comment]
Good idea. Also, thanks for testing!

**Pull request merged**